### ISSUE TITLE
How to package a workspace with Binder?

### ISSUE THREAD

**Issue by choldgraf** [jupyterlab/jupyterlab#5903]
Does anybody know of a pattern out there that gets workspaces working with Binder? So that when I click a Binder link, a particular combination of windows etc pops up? This could be a nice way to demo plugins etc

**Comment by choldgraf** [comment]
Cool - we should document that!

**Comment by ian-r-rose** [comment]
Indeed! Part of the reason I haven't yet is because it is still a bit in flux -- but that's not really a great excuse :)

**Comment by choldgraf** [comment]
One option would be to create a simple binder-examples repo purely to show off that pattern.

**Comment by ian-r-rose** [comment]
@choldgraf Here is a quick demo I put together, with some documentation about how to do it:

[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/ian-r-rose/binder-workspace-demo/master?urlpath=lab)

There is a bug we need to fix around the starting of terminals, but otherwise it should be a good place to start.

**Comment by choldgraf** [comment]
ah that's actually great haha, I didn't notice the readme :-)

Do you think it's stable enough that we could advertise on docs.mybinder.org? (or in the jupyterlab docs somewhere?)

**Comment by ian-r-rose** [comment]
Yeah, it's stable enough to advertise, I think. There are a couple minor
changes that will happen with the upcoming lab release, but the basic idea
should be the same.

**Comment by WesleyTheGeolien** [comment]
> @choldgraf the state of the art here is in the pangeo/dask example binders: you need to include (1) a [workspace template](https://github.com/dask/dask-examples/blob/master/binder/jupyterlab-workspace.json), and (2) the appropriate [`start`](https://github.com/dask/dask-examples/blob/master/binder/start) script to splice in the right `$JUPYTERHUB_USER`.
> 
> It's a little fiddly, but works.

When I opened that notebook for the first time I was mind blown thought it was awesome, especially in binders where you are trying to demonstrate a concept in a live enviroment

**Comment by ian-r-rose** [comment]
My latest commit shows a hack that fixes it temporarily, I am working on an upstream PR to fix it more correctly.
